### PR TITLE
Update github action to require valid changelog file

### PR DESCRIPTION
**Pull request by jonathansadowski**
### All Submissions:

-   [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md)?
-   [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
-   [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change?

### Changes proposed in this Pull Request:

This PR updates the GitHub action that checks the changelog to also make sure that all change files are valid.

It will check any change file in the repo for validity. Because of this, this PR also updates the change files in the repo that were previously invalid.

### How to test the changes in this Pull Request:

1. Make sure changelog validation passes on this PR.

### Other information:

-   [x] Have you added an explanation of what your changes do and why you'd like us to include them?
-   [ ] Have you written new tests for your changes, as applicable?
-   [ ] Have you successfully run tests with your changes locally?
-   [x] Have you created a changelog file for each project being changed, ie `pnpm changelog add --filter=<project>`?

### FOR PR REVIEWER ONLY:

-   [ ] I have reviewed that everything is sanitized/escaped appropriately for any SQL or XSS injection possibilities. I made sure Linting is not ignored or disabled.

### CODE REVIEW

File: plugins/woocommerce/changelog/update-store-details-task [hunk 1]
```diff
@@ -1,4 +1,4 @@
-Significance: patch
-Type: fix
+Significance: minor
+Type: add
 
-Update StoreDetails task action url to navigate to the setting page
+Add shipping-smart-defaults intro tooltips.
```
Comment (samueljseay) [inline plugins/woocommerce/changelog/update-store-details-task]: @jonathansadowski Is this change intentional? it looks the same as the changelog from `plugins/woocommerce/changelog/add-shipping-intro-tooltips`

**Review by samueljseay — DISMISSED** [review]
One of the changelog entries looks incorrect. Otherwise looks good. 🚀

File: plugins/woocommerce/changelog/update-store-details-task (hunk 1 above)
Comment (jonathansadowski) [reply plugins/woocommerce/changelog/update-store-details-task]: Thanks... nope, it wasn't... not sure what happened there... hah. Fixing!
